### ISSUE TITLE
Developer experience / "Strict mode"

### ISSUE THREAD

**Issue by MathiasKandelborg** [isaachinman/next-i18next#61]
As we all know, it's extremely nice to have all the information we need, available when coding. It saves a lot of time when we don't need to look up the possible settings or get reminded that we are using the default (not so prod-friendly) settings.

So with that in mind, I'd like to start a conversation about how the DX should be with next-18next. Next.js is already super nice when it comes to DX and that sets the bar quite high from the start - in my opinion at least.

It would be great to be able to turn on/off specific messages based on the current project. Some projects might need all their namespaces but would still want/need warnings for **x**. It's probably not that easy to implement a bunch of flags (or a great DX for that matter) so for starters, a strict mode should be sufficient. 

I'll submit a PR later, with a global strict mode setting for developer messages. Until then, I've made this issue/feature request fleshing out some details or thinking about what would improve the DX with next-i18next and make everyone's lives easier. 

Thanks for reading, have a nice day!

**Comment by isaachinman** [comment]
Great, thanks for opening this.

> I'll submit a PR later

Most likely it will be good to add a `console-message.js` util that we can reuse. I would expect that to export default a function that looks like:

```jsx
export default (messageType, message) => // do stuff
```

Wherein `messageType` is an enum of `info, warning, error`.

This is probably also an appropriate time to pull in some pretty-print package for NodeJs. It's been awhile since I looked through them, but there are several good options out there.

Let me know if this provides enough direction.

**Comment by MathiasKandelborg** [comment]
I have made a fork and some commits, for now, I've tried pretty-error which seems really nice, popular(support) and probably has the functionality needed.

I've followed your orders somewhat and created a function called createConsoleMessage which is default exported. I've also created JSDoc comments for the code, I'll just paste it here to properly explain what the function does:
```js
/**
 * Create a console log with specified type and message
 *
 * Optionally provide a traceLimit variable to increase the stack trace.
 * @param {messageTypes} messageType One of: err, warn or info
 * @param {String} message
 * @param {Number} traceLimit
 */
```

Does this satisfy the needs for a reusable console.log function? If so, I'll proceed with using this file to incorporate the strict mode setting. But, regarding that... I'm not quite sure how people should enable it. Just via the settings object on a `strictModeLogging` key?

**Comment by isaachinman** [comment]
Hi @MathiasKandelborg, thanks for getting a chance to look into this.

In general, I would prefer this pattern:

```jsx
const createConsoleMessage = (messageType, message, options) =>
```

Wherein `traceLimit` could be provided inside the options object. This will allow us to add new options in the future without having to worry about breaking changes with argument order, etc.

> I'm not quite sure how people should enable it

In my opinion, strict mode should be enabled by default, and users can opt _out_ of it. And yes, it's probably easiest to just put a bool in the config object called `strictMode`. Let me know if this makes sense. The config is already shared around to all relevant functions/components.

**Comment by isaachinman** [comment]
It would be nice not to have to pass `config.strictMode` into every single console message... But the only way I can think to do that is to really lean into this class-based approach and set `this.createConsoleMessage` on the class, so that the config is automatically accessible.

I'm really not sure if I'll continue using this class-based method in future versions of `next-i18next` - might abandon it for something more functional.

For now... I'll leave it up to you. I lean towards just putting it on the class, but if you don't like that idea, then yes - we'll need to pass `strictMode` each time.

As far as setting the `traceLimit` integer: it seems to be people are only going to be doing this in specific debugging cases, and then would immediately want it back to a lower number. Thus, I propose allowing an override with an env var, eg `NEXT_i18NEXT_TRACE_LIMIT`, or something. It might be nice to add a message to console output that lets users know how to increase the limit.

**Comment by isaachinman** [comment]
Hey @MathiasKandelborg, and happy holidays. I assume you are out of the office. If you'd like me to take care of this one, just let me know.

**Comment by MathiasKandelborg** [comment]
Hey @isaachinman! Happy holidays, I hope you had some great times and didn't blow anything up on new years eve :slightly_smiling_face: 

I've just submitted a PR with your recent additions. I'm sorry for the delay, I hope the code is fine as is. The codeclimate app seems to have caught some invalid errors. Let me know if I should fix them

**Comment by isaachinman** [comment]
Initial implementation fixed with #75.
